### PR TITLE
Version 0.19.0

### PR DESCRIPTION
**Pull request by Kludex**
## Checklist

- [x] #1640
- [x] #1671
- [x] #1661
- [x] #1645
- [x] #1652

Diff: https://github.com/encode/uvicorn/compare/0.18.3...master
Draft: https://github.com/encode/uvicorn/releases/tag/untagged-8b346412076c82a9e8cc

### CODE REVIEW

**Comment by nlsj1985** [comment]
@Kludex httptools 5.0.0 just got released with the CVE fixes in llhttp. It seems it will get included in uvicorn automatically when you create a release, but i'm not sure. Perhaps, it's best to bump the version explicitly... thanks for checking it.

**Comment by Kludex** [comment]
> @Kludex httptools 5.0.0 just got released with the CVE fixes in llhttp. It seems it will get included in uvicorn automatically when you create a release, but i'm not sure. Perhaps, it's best to bump the version explicitly... thanks for checking it.

- #1645 :+1:

**Comment by Kludex** [comment]
I'd like to make a release soon. :pray:

**Comment by nlsj1985** [comment]
note. I made a PR in the httptools project for the latest llhttp (v6.0.10) update.. but guess it's wise to first proceed with this httptools 0.5.0 in uvicorn 0.19.0 and when we get a new httptools version, queue it for the following uvicorn...

**Comment by Kludex** [comment]
Would you mind sharing the link?

**Comment by Kludex** [comment]
> note. I made a PR in the httptools project for the latest llhttp (v6.0.10) update.. but guess it's wise to first proceed with this httptools 0.5.0 in uvicorn 0.19.0 and when we get a new httptools version, queue it for the following uvicorn...

But there's nothing uvicorn needs to do... Unless the bump solved any CVE.

**Comment by nlsj1985** [comment]
httptools v5.0.0 contains llhttp release/v6.0.9, this included some fixes for the 3 CVE's that where done in llhttp release/v6.0.7

llhttp release/v6.0.10 seems to update a resolution for CVE-2022-32213:
Disable chunked on obs by @ShogunPanda in https://github.com/nodejs/llhttp/pull/196

nodejs [https://github.com/nodejs/node/releases/tag/v18.9.1](url) has a paragraph about llhttp that mentions a bit more:
_bypass via obs-fold mechanic (Medium)([CVE-2022-32213](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-32213) ): The llhttp parser in the http module does not correctly parse and validate Transfer-Encoding headers. This can lead to HTTP Request Smuggling (HRS)._

There are 3 change's in src/llhttp/http.ts:
https://github.com/nodejs/llhttp/commit/f49fbf5c818ccc7e4d9994393b8f8e1d96483d7e

So without reverse engineering the entire llhttp release/v6.0.10 commit it seemed to me there's actual improvements made for the CVE in relation to obs-fold's .. that i guess where missed in the patch for the same CVE. If you want to know more, please check the patch or ask ShokunPanda who makes the patches for the llhttp project.

I'm just a random user that's trying to be instrumental to get fixes into uvicorn asap.

**Comment by Kludex** [comment]
This is the link I wanted: https://github.com/MagicStack/httptools/pull/86 🙏 

> I'm just a random user that's trying to be instrumental to get fixes into uvicorn asap.

`uvicorn` doesn't pin `httptools`. I've only bumped the minimum requirement of `httptools` to `0.5.0` to force users to bump it, but it was actually not needed. We are only blocked by `httptools` here.

**Comment by ShogunPanda** [comment]
@nlsj1985 I can confirm folks. All OBS fold CVEs are fixed in 6.0.10

Next week I plan to release llhttp 7 with more features about passable callbacks and chunked encoding parsing, just FYI.

**Comment by Kludex** [comment]
> @nlsj1985 I can confirm folks. All OBS fold CVEs are fixed in 6.0.10
> 
> Next week I plan to release llhttp 7 with more features about passable callbacks and chunked encoding parsing, just FYI.

Thanks for coming here, and let us know. Much appreciated! 🙏

**Comment by ShogunPanda** [comment]
> > @nlsj1985 I can confirm folks. All OBS fold CVEs are fixed in 6.0.10
> 
> > 
> 
> > Next week I plan to release llhttp 7 with more features about passable callbacks and chunked encoding parsing, just FYI.
> 
> 
> 
> Thanks for coming here, and let us know. Much appreciated! 🙏 

You're welcome! 😉

**Comment by Kludex** [comment]
- Only #1706 missing for 0.19.0 to be released.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+- Support Python 3.11 (#1652) 16/09/22
+- Bump minimal `httptools` version to `0.5.0` (#1645) 13/09/22
+- Ignore HTTP/2 upgrade and optionally ignore WebSocket upgrade (#1661) 19/10/22
+- Add `py.typed` to comply with PEP 561 (#1687) 07/10/22
+
+### Fixed
+
+- Set `propagate` to `False` on "uvicorn" logger (#1288) 08/10/22
+- USR1 signal is now handled correctly on `UvicornWorker`. (#1565) 26/08/22
+- Use path with query string on `WebSockets` logs (#1385) 11/09/22
```
Comment (Kludex) [inline CHANGELOG.md:16]:
```suggestion
- Use path with query string on `WebSockets` logs (#1385) 11/09/22
- Fix behavior on which "Date" headers were the same per connection (#1706) 19/10/22 
```

**Comment by Kludex** [comment]
Ready for release. GitHub release draft ready as well.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+- Fix behavior on which "Date" headers were the same per connection (#1706) 19/10/22
```
Comment (Kludex) [inline CHANGELOG.md]:
```suggestion
- Fix behavior on which "Date" headers were not updated on the same connection (#1706) 19/10/22 
```

**Review by lovelydinosaur — APPROVED** [review]
Great. 🌟

**Comment by Kludex** [comment]
I'll release it tonight. Thanks @tomchristie 🙏

**Pull request merged**